StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/FileHeaderCodeFixProvider.cs

```diff
@@ -56,48 +57,81 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             var settings = document.Project.AnalyzerOptions.GetStyleCopSettings();
 
             var fileHeader = FileHeaderHelpers.ParseFileHeader(root);
-            var newSyntaxRoot = fileHeader.IsMissing ? AddHeader(document, root, document.Name, settings) : ReplaceHeader(document, root, settings);
+            var xmlFileHeader = FileHeaderHelpers.ParseXmlFileHeader(root);
+            var newSyntaxRoot = fileHeader.IsMissing ? AddHeader(document, root, document.Name, settings) : ReplaceHeader(document, root, settings, xmlFileHeader.IsMalformed);
```

Comment:
I'll review this once I've had a look at doing the multiline comments, but I think you're correct and two lots of code would be cleaner.